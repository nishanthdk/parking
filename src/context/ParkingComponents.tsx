import React, { createContext, useState } from "react";
import { numberofSpaceType } from "../types/ComponentsTypes";

export const ParkingContext = createContext<numberofSpaceType>({
  space: "",
  setSpace: () => {},
  registration: "",
  setRegistration: () => {},
  timings: "",
  setTimings: () => {},
});

const ParkingComponents = ({ children }: { children: React.ReactNode }) => {
  const [space, setSpace] = useState<string>("");
  const [registration, setRegistration] = useState<string>("");
  const [timings, setTimings] = useState<string>("");

  const contextValue: numberofSpaceType = {
    space,
    setSpace,
    registration,
    setRegistration,
    timings,
    setTimings,
  };

  return (
    <ParkingContext.Provider value={contextValue}>
      {children}
    </ParkingContext.Provider>
  );
};

export default ParkingComponents;
