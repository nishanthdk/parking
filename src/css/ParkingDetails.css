.container-spaces {
    max-width: 350px;
    margin-top: 10px;
    margin-left: auto;
    margin-right: auto;
    padding: 2rem;
    box-shadow: 0 2px 8px rgba(0, 0, 0, 0.2);
    /* display: flex; */
    justify-content: center;
    align-items: center;
    position: relative;
    border-radius: 10px;

  }
  
  .parking-app {
    text-align: center;
    margin-bottom: 2rem;
    color: rgb(42, 70, 87);
  }
  
  .Enter-number-of-spaces {
    font-size: 1.2rem;
    margin-right: 1rem;
  }
  
  .space-input {
    padding: 0.5rem;
    font-size: 1rem;
    border: 1px solid #ccc;
    border-radius: 4px;
    margin-right: 1rem;
    width: 60%;
    border-radius: 10px;
  }
  
  .space-submit {
    padding: 0.5rem 1rem;
    font-size: 1rem;
    background-color: #007bff;
    color: #fff;
    border: none;
    border-radius: 4px;
    cursor: pointer;
  }
  
  .space-submit:hover {
    background-color: #0069d9;
  }
  