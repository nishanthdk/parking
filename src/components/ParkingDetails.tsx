import React from "react";
import { useContext } from "react";
import { useNavigate } from "react-router-dom";
import { ParkingContext } from "../context/ParkingComponents";
import "../css/ParkingDetails.css"

const ParkingDetails = () => {
  const { space, setSpace } = useContext(ParkingContext);

  const navigate = useNavigate();

  const handleSubmit = (e: React.ChangeEvent<HTMLFormElement>) => {
      navigate("/ParkingLot")
  };

  return (
    <div className="container-spaces">
      <h1 className="parking-app">Parking App</h1>
      <form onSubmit={handleSubmit} data-testid="my-form">
        <label className="Enter-number-of-spaces">Enter the number of spaces: </label>
        <input
        className="space-input"
          data-testid="inputSpace"
          type="text"
          value={space}
          placeholder="Enter spaces"
          onChange={(e) => setSpace(e.target.value)}
          required
        />
        <button type="submit" className="space-submit" data-testid="Submit-Button-for-ParkingSpace">
          Submit
        </button>
      </form>
    </div>
  );
};

export default ParkingDetails;
