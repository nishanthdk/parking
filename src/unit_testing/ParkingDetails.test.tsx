import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import ParkingDetails from "../components/ParkingDetails";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom/extend-expect";
// import { shallow } from 'enzyme';
import { ParkingContext } from "../context/ParkingComponents";

describe("ParkingApp component", () => {
  test("renders ParkingDetails without crashing", () => {
    render(
      <MemoryRouter>
        <ParkingDetails />
      </MemoryRouter>
    );
  });

  it("renders the heading", () => {
    render(
      <MemoryRouter>
        <ParkingDetails />
      </MemoryRouter>
    );
    const heading = screen.getByText("Parking App");
    expect(heading).toBeInTheDocument();
  });
  it("renders the form", () => {
    render(
      <MemoryRouter>
        <ParkingDetails />
      </MemoryRouter>
    );

    const label = screen.getByText("Enter the number of spaces:");
    const inputField = screen.getByPlaceholderText("Enter spaces");
    const input = screen.getByTestId("inputSpace");
    const button = screen.getByTestId("Submit-Button-for-ParkingSpace");

    expect(input).toBeInTheDocument();
    expect(inputField).toBeInTheDocument();
    expect(label).toBeInTheDocument();
    expect(button).toBeInTheDocument();
  });
});
describe("ParkingApp component2", () => {
  test("handles form submission", () => {
    const setSpace = jest.fn();
    const setTimings = jest.fn();
    const setRegistration = jest.fn();
    const navigate = jest.fn();

    render(
      <MemoryRouter>
        <ParkingContext.Provider
          value={{
            space: "",
            setSpace,
            timings: "",
            setTimings,
            registration: "",
            setRegistration,
          }}
        >
          <ParkingDetails />
        </ParkingContext.Provider>
      </MemoryRouter>
    );

    const input = screen.getByTestId("inputSpace");
    fireEvent.change(input, { target: { value: "10" } });

    expect(setSpace).toHaveBeenCalledWith("10");
    const form = screen.getByTestId("my-form");
    fireEvent.submit(form);

    expect(navigate).toMatchSnapshot();
  });
});
