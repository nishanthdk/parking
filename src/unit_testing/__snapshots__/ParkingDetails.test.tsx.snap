// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParkingApp component2 handles form submission 1`] = `[MockFunction]`;
